implementations/micrometer-registry-azure-monitor/src/main/java/io/micrometer/azuremonitor/AzureMonitorMeterRegistry.java

```diff
@@ -63,11 +63,18 @@ private AzureMonitorMeterRegistry(AzureMonitorConfig config, Clock clock,
         super(config, clock);
 
         config().namingConvention(new AzureMonitorNamingConvention());
+
         if (StringUtils.isEmpty(telemetryConfiguration.getInstrumentationKey())) {
             checkRequired("instrumentationKey", AzureMonitorConfig::instrumentationKey).apply(config).orThrow();
```

Comment:
If you just configure the connection string (containing the instrumentation key) but do not configure the `instrumentationKey` on `AzureMonitorConfig`, won't this throw an error here? It seems like we should check first for and apply the connection string.

Reply:
I don't think an error would be thrown, but I pushed a new proposal relying on connection string only.